src/qtoolkit/io/sge.py

```diff
@@ -0,0 +1,526 @@
+            qjob.name = self._get_element_text(job_element, "JB_name")
+
+            info = QJobInfo()
+            info.nodes = self._safe_int(
+                self._get_element_text(job_element, "num_nodes")
+            )
+            info.cpus = self._safe_int(self._get_element_text(job_element, "num_proc"))
+            info.memory_per_cpu = self._convert_memory_str(
+                self._get_element_text(job_element, "hard resource_list.mem_free")
+            )
+            info.partition = self._get_element_text(job_element, "queue_name")
+            info.time_limit = self._convert_str_to_time(
+                self._get_element_text(job_element, "hard resource_list.h_rt")
+            )
+
+            qjob.info = info
+
+            jobs_list.append(qjob)
+
+        return jobs_list
+
+    @staticmethod
+    def _convert_str_to_time(time_str: str | None) -> int | None:
+        if time_str is None:
+            return None
+
+        parts = time_str.split(":")
+        if len(parts) == 3:
+            hours, minutes, seconds = parts
+        elif len(parts) == 2:
+            hours, minutes = "0", parts[0]
+            seconds = parts[1]
+        elif len(parts) == 1:
+            hours, minutes, seconds = "0", "0", parts[0]
+        else:
+            raise OutputParsingError(f"Invalid time format: {time_str}")
+
+        try:
+            return int(hours) * 3600 + int(minutes) * 60 + int(seconds)
+        except ValueError:
+            raise OutputParsingError(f"Invalid time format: {time_str}")
+
+    @staticmethod
+    def _convert_memory_str(memory: str | None) -> int | None:
+        if not memory:
+            return None
+
+        match = re.match(r"([0-9]+)([a-zA-Z]*)", memory)
+        if not match:
+            raise OutputParsingError("No numbers and units parsed")
+        memory, units = match.groups()
+
+        power_labels = {"K": 0, "M": 1, "G": 2, "T": 3}
+
+        if not units:
+            units = "M"
+        elif units not in power_labels:
+            raise OutputParsingError(f"Unknown units {units}")
+        try:
+            v = int(memory)
+        except ValueError:
+            raise OutputParsingError
+
+        return v * (1024 ** power_labels[units.upper()])
+
+    _qresources_mapping = {
+        "queue_name": "queue",
+        "job_name": "job_name",
+        "account": "account",
+        "priority": "priority",
+        "output_filepath": "qout_path",
+        "error_filepath": "qerr_path",
+        "project": "group_list",
+    }
+
+    @staticmethod
+    def _convert_time_to_str(time: int | float | timedelta) -> str:
+        if not isinstance(time, timedelta):
+            time = timedelta(seconds=time)
+
+        hours, remainder = divmod(int(time.total_seconds()), 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        time_str = f"{hours}:{minutes}:{seconds}"
+        return time_str
+
+    def _convert_qresources(self, resources: QResources) -> dict:
```

Comment:
I put this comment here, but could be a more general one. 
Given the similarities between SGE and PBS (expecially this method is almost identical), do you think it would be worth to have a common base class?

Reply:
actually I'd say that this is a very reasonable thing to do. What would you say is the best approach? make a common.py file or edit the functions in base.py? Or maybe something else?

Reply:
This sound very good to me. I will implement it like that then.

Reply:
I have implemented this now and the sge unit tests pass. I just needed to adjust the unit tests because of the softwalltime factor and because I changed _get_jobs_list_cmd because it is actually possible in GSE to query by job_list as I mentioned here https://github.com/Matgenix/jobflow-remote/pull/187#issuecomment-2363451641 .
So I would say it should continue to work for the PBS part as well. As there are no PBS unit tests, I cannot verify this though.